### PR TITLE
Fix Ledger close_time_iso 

### LINKED ISSUE #1007: close_time_iso  in transactions  of ledger is different from rippled
Request:

`{"method": "ledger", "params": [{"transactions": true, "expand": true, "owner_funds": false, "ledger_index": 2034680, "api_version": 2, "Fee": 20}]}`

Rippled response:

`{
    "result": {
        "ledger": {
            "account_hash": "D8E906E135B9D959B16C2A2E9CFB7E356AA9C3D56AEE024C196AF6D3A77AEE4E",
            "close_flags": 0,
            "close_time": 754531472,
            "close_time_human": "2023-Nov-29 00:04:32.000000 UTC",
            "close_time_iso": "2023-11-29T00:04:32Z",
            "close_time_resolution": 10,
            "closed": true,
            "ledger_hash": "6703F2D4778C39C87F9FB49722387E9A401B07FCC5398901214B7A7792A70003",
            "ledger_index": "2034680",
            "parent_close_time": 754531471,
            "parent_hash": "EDD1227C7C094CB1FD34EB900507C3FC14F699F2B2567B7AD6E027D71C8DF727",
            "total_coins": "99999319285186216",
            "transaction_hash": "911E9A44DBD15368C7CAB1D0D3A4678B5A9C405C4AF1139D5E469DEA7F036052",
            "transactions": [
                {
                    "close_time_iso": "2023-11-29T00:04:32Z",
                    "hash": "98B0B460460F6F78E201946BFD7BD4087E646B98E1D87DF0ED6C56B866D75207",
                    "ledger_hash": "6703F2D4778C39C87F9FB49722387E9A401B07FCC5398901214B7A7792A70003",
                    "ledger_index": "2034680",
                    "meta": {
                        "AffectedNodes": [
                            {
                                "CreatedNode": {
                                    "LedgerEntryType": "AccountRoot",
                                    "LedgerIndex": "17745CA915B76CDB68EFF6C64900F0BD3CA4057F06CAD6B08B968EC071AB64DA",
                                    "NewFields": {
                                        "Account": "rnYL9GBjQXi8LPDUQSD8fwhfsWFE1UWhpz",
                                        "Balance": "40000000",
                                        "Sequence": 2034680
                                    }
                                }
                            },
                            {
                                "ModifiedNode": {
                                    "FinalFields": {
                                        "Account": "rLGfF122ZHdmfqw6PYTMogJMKzPpgYcrw4",
                                        "Balance": "19986429993300",
                                        "Flags": 0,
                                        "OwnerCount": 0,
                                        "Sequence": 2034038
                                    },
                                    "LedgerEntryType": "AccountRoot",
                                    "LedgerIndex": "1E7B0B0BBCCFE8DB6EAC44A4DEFC0E89D3A12DDA5B8BE180A143F1B92C5069BB",
                                    "PreviousFields": {
                                        "Balance": "19986469993320",
                                        "Sequence": 2034037
                                    },
                                    "PreviousTxnID": "E6E055F9696607F966E70C520FC4224C6DC8CFC4722F17E5CF832D2E06A2119C",
                                    "PreviousTxnLgrSeq": 2034676
                                }
                            }
                        ],
                        "TransactionIndex": 0,
                        "TransactionResult": "tesSUCCESS",
                        "delivered_amount": "40000000"
                    },
                    "tx_json": {
                        "Account": "rLGfF122ZHdmfqw6PYTMogJMKzPpgYcrw4",
                        "DeliverMax": "40000000",
                        "Destination": "rnYL9GBjQXi8LPDUQSD8fwhfsWFE1UWhpz",
                        "Fee": "20",
                        "Flags": 2147483648,
                        "Sequence": 2034037,
                        "SigningPubKey": "02931E8557368B801C0FFD9398A1B08B0544EC035E6820B8496D9962B704F2C79E",
                        "TransactionType": "Payment",
                        "TxnSignature": "3045022100C494F8551AC1D6A305B745A53357BC3B1FD339AB3903034A94F42B8CBCC6F5AE022062D40A027146A6E4AA7A8D64349EEF67ECA8F26D645C38289F958FFC62A38B21"
                    },
                    "validated": true
                }
            ]
        },
        "ledger_hash": "6703F2D4778C39C87F9FB49722387E9A401B07FCC5398901214B7A7792A70003",
        "ledger_index": 2034680,
        "status": "success",
        "validated": true
    }
}`

Clio:
`{
    "result": {
        "ledger_hash": "6703F2D4778C39C87F9FB49722387E9A401B07FCC5398901214B7A7792A70003",
        "ledger_index": 2034680,
        "validated": true,
        "ledger": {
            "account_hash": "D8E906E135B9D959B16C2A2E9CFB7E356AA9C3D56AEE024C196AF6D3A77AEE4E",
            "close_flags": 0,
            "close_time": 754531472,
            "close_time_human": "2023-Nov-29 00:04:32.000000000 UTC",
            "close_time_resolution": 10,
            "close_time_iso": "2023-11-29T00:04:32Z",
            "ledger_hash": "6703F2D4778C39C87F9FB49722387E9A401B07FCC5398901214B7A7792A70003",
            "ledger_index": "2034680",
            "parent_close_time": 754531471,
            "parent_hash": "EDD1227C7C094CB1FD34EB900507C3FC14F699F2B2567B7AD6E027D71C8DF727",
            "total_coins": "99999319285186216",
            "transaction_hash": "911E9A44DBD15368C7CAB1D0D3A4678B5A9C405C4AF1139D5E469DEA7F036052",
            "closed": true,
            "transactions": [
                {
                    "validated": true,
                    "ledger_index": "2034680",
                    "close_time_iso": "2023-11-28T19:10:11Z",
                    "ledger_hash": "6703F2D4778C39C87F9FB49722387E9A401B07FCC5398901214B7A7792A70003",
                    "hash": "98B0B460460F6F78E201946BFD7BD4087E646B98E1D87DF0ED6C56B866D75207",
                    "tx_json": {
                        "Account": "rLGfF122ZHdmfqw6PYTMogJMKzPpgYcrw4",
                        "DeliverMax": "40000000",
                        "Destination": "rnYL9GBjQXi8LPDUQSD8fwhfsWFE1UWhpz",
                        "Fee": "20",
                        "Flags": 2147483648,
                        "Sequence": 2034037,
                        "SigningPubKey": "02931E8557368B801C0FFD9398A1B08B0544EC035E6820B8496D9962B704F2C79E",
                        "TransactionType": "Payment",
                        "TxnSignature": "3045022100C494F8551AC1D6A305B745A53357BC3B1FD339AB3903034A94F42B8CBCC6F5AE022062D40A027146A6E4AA7A8D64349EEF67ECA8F26D645C38289F958FFC62A38B21"
                    },
                    "meta": {
                        "AffectedNodes": [
                            {
                                "CreatedNode": {
                                    "LedgerEntryType": "AccountRoot",
                                    "LedgerIndex": "17745CA915B76CDB68EFF6C64900F0BD3CA4057F06CAD6B08B968EC071AB64DA",
                                    "NewFields": {
                                        "Account": "rnYL9GBjQXi8LPDUQSD8fwhfsWFE1UWhpz",
                                        "Balance": "40000000",
                                        "Sequence": 2034680
                                    }
                                }
                            },
                            {
                                "ModifiedNode": {
                                    "FinalFields": {
                                        "Account": "rLGfF122ZHdmfqw6PYTMogJMKzPpgYcrw4",
                                        "Balance": "19986429993300",
                                        "Flags": 0,
                                        "OwnerCount": 0,
                                        "Sequence": 2034038
                                    },
                                    "LedgerEntryType": "AccountRoot",
                                    "LedgerIndex": "1E7B0B0BBCCFE8DB6EAC44A4DEFC0E89D3A12DDA5B8BE180A143F1B92C5069BB",
                                    "PreviousFields": {
                                        "Balance": "19986469993320",
                                        "Sequence": 2034037
                                    },
                                    "PreviousTxnID": "E6E055F9696607F966E70C520FC4224C6DC8CFC4722F17E5CF832D2E06A2119C",
                                    "PreviousTxnLgrSeq": 2034676
                                }
                            }
                        ],
                        "TransactionIndex": 0,
                        "TransactionResult": "tesSUCCESS",
                        "delivered_amount": "40000000"
                    }
                }
            ]
        },
        "status": "success"
    },
    "warnings": [
        {
            "id": 2001,
            "message": "This is a clio server. clio only serves validated data. If you want to talk to rippled, include 'ledger_index':'current' in your request"
        }
    ]
}`

### PR DESCRIPTION
**Pull request by cindyyan317**
Fix #1007 
Misusage of static variable in lambda

### CODE REVIEW

File: src/rpc/handlers/Ledger.cpp [hunk 1]
```diff
@@ -79,8 +79,9 @@ LedgerHandler::process(LedgerHandler::Input input, Context const& ctx) const
                 return toJsonWithBinaryTx(tx, ctx.apiVersion);
             };
 
+            auto const isoTimeStr = ripple::to_string_iso(lgrInfo.closeTime);
```
Comment (kuznetsss) [inline src/rpc/handlers/Ledger.cpp:82]: Could you explain what is the difference except optimisation?
`lgrInfo` is `const` so it shouldn't be different in every call of this lambda.

Comment (kuznetsss) [reply src/rpc/handlers/Ledger.cpp:82]: I see, it was `static`

**Review by godexsoft — APPROVED** [review]
LGTM 🚀

**Review by kuznetsss — APPROVED** [review]
LGTM

**Pull request merged**